arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,391 @@
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    func addCondition() {
+        if configuration == nil {
+            configuration = AGSUtilityTraceConfiguration()
+        }
+        if configuration?.traversability == nil {
+            configuration?.traversability = AGSUtilityTraversability()
+        }
+        // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator.
+        if selectedAttribute != nil {
+            // If the value is a coded value.
+            if let codedValue = selectedValue as? AGSCodedValue, selectedAttribute?.domain is AGSCodedValueDomain {
+                selectedValue = convertToDataType(otherValue: codedValue.code!, dataType: selectedAttribute!.dataType)
+            } else {
+                selectedValue = convertToDataType(otherValue: selectedValue!, dataType: selectedAttribute!.dataType)
+            }
+            // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+            var expression: AGSUtilityTraceConditionalExpression?
+            expression = AGSUtilityNetworkAttributeComparison(networkAttribute: selectedAttribute!, comparisonOperator: selectedComparison!, value: selectedValue!)
+            if let otherExpression = configuration?.traversability?.barriers as? AGSUtilityTraceConditionalExpression {
+                // NOTE: You may also combine expressions with UtilityTraceAndCondition
+                expression = AGSUtilityTraceOrCondition(leftExpression: otherExpression, rightExpression: expression!)
+            }
+            // Update the list of barrier conditions.
+            configuration?.traversability?.barriers = expression
+            let expressionString = expressionToString(expression: expression!)
+            textView?.text += "\n \(expressionString!))"
+        }
+    }
+    
+    func reset() {
+        // Reset the barrier condition to the initial value.
+        let traceConfiguration = configuration
+        traceConfiguration?.traversability?.barriers = initialExpression
+        textView?.text = expressionToString(expression: initialExpression!)
+        // Reset the selected values.
+        selectedAttribute = nil
+        selectedComparison = nil
+        selectedValue = nil
+        //Reset the labels.
+        attributeLabel.text = nil
+        comparisonLabel.text = nil
+        valueLabel.text = nil
+    }
+    
+    func trace() {
+        if utilityNetwork == nil || startingLocation == nil {
+            presentAlert(title: "Error", message: "Could not trace utility network.")
+            return
+        } else {
+            // Create utility trace parameters for the starting location.
+            let startingLocations = [startingLocation]
+            let parameters = AGSUtilityTraceParameters(traceType: .subnetwork, startingLocations: startingLocations as! [AGSUtilityElement])
```

Comment:
```suggestion
            let parameters = AGSUtilityTraceParameters(traceType: .subnetwork, startingLocations: [startingLocation])
```
Maybe do not need casting here.

Reply:
It produced an error without the cast I think because it is declared globally on line 65 as an optional.